rtl/eclrtl/rtldynfield.cpp

```diff
@@ -510,3 +510,480 @@ extern ECLRTL_API int getFieldNum(const char *fieldName, IOutputMetaData &  meta
+                    break;
+                }
+                case match_typecast:
+                    offset = translateScalar(builder, offset, field, type, sourceType, source);
+                    break;
+                case match_link:
+                {
+                    // a 32-bit record count, and a (linked) pointer to an array of record pointers
+                    byte *dest = builder.ensureCapacity(offset+sizeof(size32_t)+sizeof(byte **), field->name)+offset;
+                    *(size32_t *)dest = *(size32_t *)source;
+                    *(byte ***)(dest + sizeof(size32_t)) = rtlLinkRowset(*(byte ***)(source + sizeof(size32_t)));
+                    offset += sizeof(size32_t)+sizeof(byte **);
+                    break;
+                }
+                case match_recurse:
+                    if (type->getType()==type_record)
+                        offset = match.subTrans->translate(builder, offset, source);
+                    else if (type->isLinkCounted())
+                    {
+                        // a 32-bit record count, and a pointer to an array of record pointers
+                        IEngineRowAllocator *childAllocator = builder.queryAllocator()->createChildRowAllocator(type->queryChildType());
+                        assertex(childAllocator);  // May not be available when using serialized types (but unlikely to want to create linkcounted children remotely either)
+
+                        size32_t sizeInBytes = sizeof(size32_t) + sizeof(void *);
+                        builder.ensureCapacity(offset+sizeInBytes, field->name);
+                        size32_t numRows = 0;
+                        byte **childRows = nullptr;
+                        if (sourceType->isLinkCounted())
+                        {
+                            // a 32-bit count, then a pointer to the source rows
+                            size32_t childCount = *(size32_t *) source;
+                            source += sizeof(size32_t);
+                            const byte ** sourceRows = *(const byte***) source;
+                            for (size32_t childRow = 0; childRow < childCount; childRow++)
+                            {
+                                RtlDynamicRowBuilder childBuilder(*childAllocator);
+                                size32_t childLen = match.subTrans->translate(childBuilder, 0, sourceRows[childRow]);
+                                childRows = childAllocator->appendRowOwn(childRows, ++numRows, (void *) childBuilder.finalizeRowClear(childLen));
+                            }
+                        }
+                        else
+                        {
+                            // a 32-bit size, then rows inline
+                            size32_t childSize = *(size32_t *) source;
+                            source += sizeof(size32_t);
+                            const byte *initialSource = source;
+                            while (source - initialSource < childSize)
```

Comment:
a pointer difference has type ptrdiff_t - which is signed.  Techincally it probably needs casting to size_t.  It probably only generates a warning on a 32bit build.